File: Problem-055/solution.py
def count_lychrel_numbers(num):

    count = 0

    for i in range(1, num):

        temp = i
        is_ly = True

        for j in range(50):

            temp = temp + reversed_num(temp)

            if pall(temp):
                is_ly = False
                break

        if is_ly:
            count += 1

    return count


def reversed_num(n):

    return int(str(n)[::-1])


def pall(n):

    return n == reversed_num(n)


print(count_lychrel_numbers(10000))

File: Problem-055/solution.js
function countLychrelNumbers(num) {
  let count = 0;
  for (let i = 1; i< num ; i++) {
    let temp = i;
    let isLy = true;
    for (let j= 0; j<50; j++){
      temp = temp + reversed(temp);
      if ( pall(temp)){
        isLy =false;
        break;
      }
    }
    if (isLy) count++;
  }
  return count;
}
function reversed(n){
  return Number(n.toString().split("").reverse().join(''));
}
function pall(n){
  return n === reversed(n);
}
